examples/agentInteractionsDemo/app.js

```diff
@@ -175,13 +175,88 @@ new Vue({
+            return newConv.customer
+        }
+
+        async function getCustomerName(customer){
+            if (customer.externalContactId){
+                return externalContactsApi.getExternalcontactsContact(customer.externalContactId)
+                    .then((externalContact) => {
+                        const name = `${externalContact.firstName} ${externalContact.lastName}`;
+                        return Promise.resolve(name)
+                    })
+            } else{
+                if(!customer.name){
+                    return Promise.resolve("Unknown")
+                } else{
+                    return Promise.resolve(customer.name)
+                }
+            } 
+        }
+
+        async function getConversationData(convId){
+            return conversationsApi.getConversation(convId).then(
+                    async (conv) => {
+                        conv.customer = getCustomerParticipant(conv);
+                        const name = await getCustomerName(conv.customer);
+                        conv.customer.name = name;
+                        return conv;
+                    }
+                ).catch((err) => {
+                    console.log(`Error: ${err}`);
+                })
+        }
+
+        async function getConversationsAndEvaluations(agentUserId) {
+            const startTime = moment('1997-01-01').toISOString();
+            const endTime = moment().toISOString();
+            return getEvaluations(qualityApi, startTime, endTime, agentUserId)
+                .then(async (data) => {
+                    const evaluations = data.entities;
+                    const evalConversations = {}
+
+                    return Promise.all(
+                        evaluations.map(async eval => {
+                            const evalConvId = eval.conversation.id
+                            if (!(evalConvId in evalConversations)){
+                                evalConversations[evalConvId] = {evals: []}
+                            }
+                            evalConversations[evalConvId].evals.push(eval)
+                            const evalConv = await getConversationData(evalConvId);
+                            return evalConversations[evalConvId].conv = evalConv;
+                        })
+                    ).then(() => {return evalConversations})
+
+                }).then(async (conversations) => {
+                    return conversationsApi.getConversations().then(async (data)=> {
+                        await Promise.all(
+                            data.entities.filter(conv => !(conv.id in conversations))
+                                .map(conv_1 => {
+                                    conversations[conv_1.id] = {
+                                        evals: []
+                                    };
+                                    return getConversationData(conv_1.id).then(newConv => conversations[newConv.id].conv = newConv);
```

Comment:
where you seeing cases where the conversation data payload differed from getConversations and getConversation?

Reply:
I should have named my variables better, sorry. Basically, this Promise.all gets all the conversations while the previous one only gets the ones with evaluations. So I'm just filtering out the conversations that we already got from the previous Promise.all. Is that what your question was asking about?